colorbleed/plugins/maya/load/load_yeti_cache.py

```diff
@@ -83,30 +84,88 @@ def remove(self, container):
+        # Get all node data
+        fname, ext = os.path.splitext(path)
+        settings_fname = "{}.fursettings".format(fname)
+        with open(settings_fname, "r") as fp:
+            settings = json.load(fp)
+
+        # Collect scene information of asset
+        set_members = cmds.sets(container["objectName"], query=True)
+        container_root = lib.get_container_transforms(container,
+                                                      members=set_members,
+                                                      root=True)
+        scene_nodes = cmds.ls(set_members, type="pgYetiMaya", long=True)
+
+        # Build lookup with cbId as keys
+        scene_lookup = defaultdict(list)
+        for node in scene_nodes:
+            cb_id = lib.get_id(node)
+            scene_lookup[cb_id].append(node)
+
+        # Re-assemble metadata with cbId as keys
+        meta_data_lookup = {n["cbId"]: n for n in settings["nodes"]}
+
+        # Compare look ups and get the nodes which ar not relevant any more
+        to_delete_lookup = {cb_id for cb_id in scene_lookup.keys() if
+                            cb_id not in meta_data_lookup}
+        if to_delete_lookup:
+
+            # Get nodes and remove entry from lookup
+            to_remove = []
+            for _id in to_delete_lookup:
+                # Get all related nodes
+                shapes = scene_lookup[_id]
+                # Get the parents of all shapes under the ID
+                transforms = cmds.listRelatives(shapes, parent=True,
+                                                fullPath=True) or []
+                to_remove.extend(shapes + transforms)
+
+                # Remove id from look uop
+                scene_lookup.pop(_id, None)
+
+            cmds.delete(to_remove)
+
+        for cb_id, data in meta_data_lookup.items():
+
+            # Update cache file name
+            file_name = data["name"].replace(":", "_")
+            cache_file_path = "{}.%04d.fur".format(file_name)
+            data["attrs"]["cacheFileName"] = os.path.join(path, cache_file_path)
+
+            if cb_id not in scene_lookup:
+                new_nodes = self.create_nodes(namespace, [data])
+                cmds.sets(*new_nodes, addElement=container_node)
+                cmds.parent(new_nodes, container_root)
+
+            else:
+                # Update the matching nodes
+                scene_nodes = scene_lookup[cb_id]
+                match_name = meta_data_lookup[cb_id]["name"]
+
+                for scene_node in scene_nodes:
+                    # Remove the namespace root ":namespace:"
+                    compare_name = scene_node.split(namespace)[-1][1:]
+                    if compare_name != match_name:
+                        self.log.info("Renaming %s to %s" %
+                                      (scene_node, match_name))
+
+                        # Get transform node, this makes renaming easier
+                        transform_node = scene_node.rstrip("Shape")
```

Comment:
Let's do `maya.cmds.listRelatives` similar to [here](https://github.com/Colorbleed/colorbleed-config/pull/139/files#diff-e54e80d1c1b4885422ca99bcc6b69d44R124)